seshat/apps/core/static/core/js/map_functions.js

```diff
@@ -402,4 +396,17 @@ function longAbsentPresentVarName(var_name){
         var_name = `${var_name[0].toUpperCase()}${var_name.slice(1)}`;
     }
     return var_name;
+}
+
+function shouldDisplayComponent(displayComponent, shape) {
```

Comment:
This reminds me that we probably want to write docstrings for all the JavaScript functions as well, even though Sphinx doesn't automatically generate doc files for that in the automatic rendering of the documentation. It can be a separate issue, and we can do that as we're refactoring the codebase. I think there's some embedded JavaScript in some of the HTML files that we want to extract into separate `.js` documents.

Reply:
You're right, I also haven't attempted to tackle testing for the JS functions yet either - see https://github.com/edwardchalstrey1/seshat/issues/119 - you're right that some of the functions are embedded in the template (where that's the only place it's used) but there's also a script where a lot of the map JS functions live - would be a good idea to refactor these

Reply:
Yep! I'll open a separate issue for it!